src/hooks/useArNSRecords.tsx

```diff
@@ -0,0 +1,40 @@
+import { AoArNSNameData } from '@ar.io/sdk/web';
+import { useGlobalState } from '@src/state';
+import { useQuery } from '@tanstack/react-query';
+
+export function useArNSRecords({
+  filters,
+}: {
+  filters?: Partial<
+    Record<
+      keyof AoArNSNameData,
+      string | number | boolean | string[] | number[] | boolean[]
+    >
+  >;
+} = {}) {
+  const [{ arioContract }] = useGlobalState();
+  return useQuery({
+    queryKey: ['arns-records', arioContract.process.processId, filters],
```

Comment:
Good hook - however this is where managing the query cache becomes weird.

Right now we have multiple records hooks fetching records. useArNSRecord, and useArNSRecords.

useArNSRecords keys by filters and useArNSRecord keys by the arns name.

.... meaning we may have the data but we will refetch it just because the predicate for the data doesn't match up.

Im going to investigate some patterns we can use for interacting with our api's and caches and follow up in dm's for options.